### PR TITLE
[4687] - Internal server error on returning to In-Store delivery from Billing - fixed

### PR DESCRIPTION
**Pull request by leonazava**
**Related issue(s):**
* Fixes scandipwa/scandipwa/issues/4687

**Problem:**
* On returning from billing, specifically in the case of returning to In-Store Pickup store selection view, the selectedShippingMethod prop would reset to an empty object which in turn would get passed to the CheckoutBilling and cause the internal server error.

**In this PR:**
* Made the CheckoutChipping.container set selectedShippingMethod to in-store pickup when on mount, isPickInStoreMethodSelected is true (In-Store Pickup is selected from the start)
* Since In-Store Pickup doesn't require customer address, removed the `isRequired ` tag from newShippingId (this fixes a propTypes error)
* Changed selectedPaymentCode state property from null to an empty string to bring it in line with its PropType definition.

### CODE REVIEW

File: packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js [hunk 1]
```diff
@@ -105,27 +106,29 @@ export class CheckoutShippingContainer extends PureComponent {
...
-        if (shippingMethods.find(({ method_code }) => method_code === selectedMethodCode)) {
-            return;
-        }
+    componentDidUpdate(prevProps) {
+        const { isPickInStoreMethodSelected, shippingMethods = [] } = this.props;
+        const { isPickInStoreMethodSelected: prevIsPickInStoreMethodSelected } = prevProps;
 
-        const [defaultShippingMethod] = shippingMethods.filter((method) => method.available);
-        const selectedShippingMethod = defaultShippingMethod || {};
+        const instore = shippingMethods.find((el) => el.method_code === STORE_IN_PICK_UP_METHOD_CODE);
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js:122]: This code is repeated with the code in the `componentDidMount`. Please move this to a separate method and reuse it.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js (hunk 1 above)
Comment (leonazava) [reply packages/scandipwa/src/component/CheckoutShipping/CheckoutShipping.container.js]: done

**Review by zans-laksa — APPROVED** [review]

**Comment by aleksandrakorolova** [comment]
Ooops page appears when opening Pick in-store tab
Error message 'TypeError: Cannot read properties of undefined (reading 'method_code')' appears in the console
Screenshot 2022-04-25 at 15 20 25
**Steps:**
1.Login with user who has product to cart
2.Go to checkout
3.Click on button Pick in store
or
1. Login with user who doesn't have products to cart or don't login at all
2. Add product to cart
3. Open Checkout
4. Click on button Pick a store
5. Open another page - PLP/PDP/CMS page
6. Return to the checkout
7. Click on button Pick in store

**Review by zans-laksa — APPROVED** [review]

**Comment by aleksandrakorolova** [comment]
Internal server error appears when going to the billing step with the same steps as previous comment 
debugMessage: "The shipping information was unable to be saved. Error: \"The address failed to save. Verify the address and try again.\""
Screenshot 2022-04-27 at 13 19 21 (2)

**Steps:**
1. Open website 
2. Don't matter user logged in or not
3. Add product to cart
4. Open Checkout
5. Click on button Pick a store
6. Select Store
7. Open another page - PLP/PDP/CMS page
8. Return to the checkout
9. Click on button Pick in store
10. Click on Proceed to Billing

**Notes:**
Is user again selects Store => billing step opens

**Comment by aleksandrakorolova** [comment]
'Undefined, Undefined' is displayed instead of 'We could not preselect pickup location based on available information, please select it manually.' for few sec + button 'Proceed to Billing' is active for this time with the same steps as in previous comment. Please, fix it too
https://drive.google.com/file/d/14sKCkHNDjEjE9_BSdu2pBCuJDAXpjZCs/view?usp=sharing

**Pull request merged**